pkg/cloud/cloud.go

```diff
@@ -1527,20 +1531,27 @@ func (c *cloud) listSnapshots(ctx context.Context, request *ec2.DescribeSnapshot
 }
 
 // waitForVolume waits for volume to be in the "available" state.
-// On a random AWS account (shared among several developers) it took 4s on average.
 func (c *cloud) waitForVolume(ctx context.Context, volumeID string) error {
+	// Based on our testing in us-west-2 and ap-south-1, the median/p99 time until volume creation is ~1.5/~4 seconds.
+	// We have found that the following parameters are optimal for minimizing provisioning time and DescribeVolumes calls
+	// we queue DescribeVolume calls after [1.25, 0.5, 0.75, 1.125, 1.7, 2.5, 3.8, 5.7, 8.6, 12.8, 19.2] seconds.
+	// In total, we wait for ~60 seconds.
 	var (
-		checkInterval = 3 * time.Second
-		// This timeout can be "ovewritten" if the value returned by ctx.Deadline()
-		// comes sooner. That value comes from the external provisioner controller.
-		checkTimeout = 1 * time.Minute
+		initialDelay = volumeCreationPollInitialDelay
+		backoff      = wait.Backoff{
+			Duration: volumeCreationPollBackoffDuration,
+			Factor:   volumeCreationPollBackoffFactor,
+			Steps:    volumeCreationPollBackoffSteps,
+		}
 	)
 
+	time.Sleep(initialDelay)
```

Comment:
Can we remove this `time.Sleep` altogether? at least for the scenario when batching is enabled?

Ran 1k volume scalability tests today and the data suggests we should do just that.

- v1.30.0 
```
+-------------------------------------------------------------+
|                     EC2 API Calls Table                     |
+-------------------+-------+------+--------------+-----------+
|               API | Count |   OK | Client Error | Throttled |
+-------------------+-------+------+--------------+-----------+
|      CreateVolume |  1000 | 1000 |            0 |         0 |
|   DescribeVolumes |   389 |  389 |            0 |         0 |
+-------------------+-------+------+--------------+-----------+
```

- This PR **without** an initial delay
```
+-------------------------------------------------------------+
|                     EC2 API Calls Table                     |
+-------------------+-------+------+--------------+-----------+
|               API | Count |   OK | Client Error | Throttled |
+-------------------+-------+------+--------------+-----------+
|      CreateVolume |  1000 | 1000 |            0 |         0 |
|   DescribeVolumes |   394 |  394 |            0 |         0 |
+-------------------+-------+------+--------------+-----------+
```



I was scratching my head at these results, and then realized that we batch requests for a total of 1 second before making the DV call. Thus, our current assumption that the first DV call is guaranteed to fail is wrong. 

Besides this request, pr LGTM.

Reply:
See the note in the PR

> NOTE: Tested with 500ms maxDelay (instead of 1s) for describeVolume batcher, because we will decrease that value in a future parameter tuning PR. That maxDelay value mostly affected p90 p95 (I presume those affected were the first DescribeVolume calls in every batch)

--- 
> ...we batch requests for a total of 1 second before making the DV call. Thus, our current assumption that the first DV call is guaranteed to fail is wrong.

I don't think this assertion is true. What about all the DV calls that are queued up in batcher within that 1 second? Some of those DV calls might be made 0.1s after createVolume succeeds, and those will likely fail. (Because we know it takes at least 1.4 seconds for a volume to be created). 

MaxDelay in a batcher adds variance to how long we wait before making the DV call, and I don't think we should rely on that as the initialDelay for any polling we do. The larger the maxDelay, the larger the variance. 

Furthermore DV is called in other places such as the ModifyVolume path. A higher maxDelay adds latency for those RPCs. 

Finally, what if we're describing more than 500 volumes within a 1 second interval? Suddenly all the DV calls don't wait the rest of that delay interval, and are more likely to fail & retry.

Reply:
Although you do bring up a great point of checking if batching is enabled. We should consider a longer delay in that case.

Reply:
> Because we know it takes at least 1.4 seconds for a volume to be created

Do we know this? I was under the impression that testing suggested 1.5 seconds was the _median_ time to create a volume, not minimum. (On that note, assuming I'm correct - do we have a minimum time to create a volume?)

Reply:
Discussed offline: I am fine with leaving it at 1.25 seconds based on the data we have.